include/eve/detail/apply_over.hpp

```diff
@@ -18,25 +18,32 @@ namespace eve::detail
 {
   // -----------------------------------------------------------------------------------------------
   // N parameters apply_over
-  template<typename Obj, value T0, value ... T>
-  EVE_FORCEINLINE  auto apply_over(Obj f
-                                  , T0 const & arg0
-                                  , T const &... args ) noexcept
-  requires simd_value<T0> || (simd_value<T> ||  ...)
+  template<typename Obj, value T0, value... T>
+  EVE_FORCEINLINE auto apply_over(Obj f, T0 const& arg0, T const&... args) noexcept
+    requires simd_value<T0> || (simd_value<T> || ...)
   {
     constexpr bool any_aggregated = has_aggregated_abi_v<T0> || (has_aggregated_abi_v<T> || ...);
     constexpr bool any_emulated   = has_emulated_abi_v<T0>   || (has_emulated_abi_v<T>   || ...);
-    if constexpr(any_aggregated) return aggregate(f, arg0, args...);
-    else if constexpr(any_emulated)   return map(f, arg0, args...);
-    else                              return f(arg0, args...);
 
+    if      constexpr(any_aggregated) return aggregate(f, arg0, args...);
+    else if constexpr(any_emulated)
+    {
+      using r_e = decltype(f(std::declval<element_type_t<T0>>(), std::declval<element_type_t<T>>()...));
+      using r_t = as_wide_t<r_e, fixed<std::max({cardinal_t<T0>{}, cardinal_t<T>{}...})>>;
```

Comment:
Some requirements/code mismatch.

If `requires simd_value<T0>` - then we know for a fact that all values better have cardinal_v<T0>. If it can be scalar - then the requirement is incorrect.

Reply:
I don't understand your remark.